src/js/view/view.js

```diff
@@ -340,9 +340,49 @@ function View(_api, _model) {
         playerViewModel.change('playlistItem', onPlaylistItem);
         // Triggering 'resize' resulting in player 'ready'
         _lastWidth = _lastHeight = null;
+
+        // Setup floating scroll handler
+        if (_floatingConfig && OS.mobile) {
+            viewsManager.addScrollHandler(throttledFloatScrollHandler);
+        }
+
         this.checkResized();
     };
 
+    // Functions for handler float on scroll (mobile)
+    const topOffset = 62;
+    let canFire = true;
+    let debounceTO;
+    function checkFloatOnScroll() {
+        const floating = _model.get('isFloating');
+        const hasCrossedThreshold = playerBounds.top < topOffset ?
```

Comment:
Is there a condition where we should exit early or always stop floating?
Examples:
- player is not in DOM
- bounds are empty (width or height === 0)

And should this function be invoked when inDOM or bounds change?

Reply:
we remove the scroll handler when the player is destroyed so that takes care of #1 (other cleanup handled by other code) - for point #2 what function would you suggest to put that call into? it's a good catch, we should recheck when bounds change

Reply:
1. A player instance can be removed (or not added) to the DOM without being destroyed (See `_model.set('inDom', inDOM);` in view.js).
2. One that doesn't cause recursion 😱 `updateBounds` is where we actually set `playerBounds`, but `checkResized` or `updateStyles` is where we usually want to respond to those changes. Using `throttledFloatScrollHandler` would be good so we're not over doing it 😌

Reply:
Let's chat monday, not totally clear on the second - are you suggesting adding a call to the throttled handler to `updateBounds`? just wanna be sure i'm clear before i implement!

Reply:
Consider:
```js
    function _responsiveUpdate() {
        if (!_this.isSetup) {
            return;
        }
        _this.updateBounds();
        _this.updateStyles();
        _this.checkResized();
    }
```
`playerBounds` is changed in `updateBounds`, classes and styles are then modified in `updateStyles` and events are dispatched in `checkResized`. When would be the best time to run `throttledFloatScrollHandler` in a responsive update or `resize` call?

Reply:
Given bounds updates are really what we want to respond to and we are only setting/removing styles in the case where we _are_ floating it seems like the right call to do it in `updateStyles` so we can update all the styles in one function call. going to put it there, let me know what you think